### PR TITLE
Mlo unity sdk

### PR DESCRIPTION
**Pull request by lgarner**
Optional device interface at constructor, and API DME optionals. That's it.

### CODE REVIEW

File: Runtime/Scripts/MobiledgeXIntegrationHelper.cs [hunk 1]
```diff
@@ -106,13 +126,12 @@ public async Task<bool> Register()
         /// To use Performance mode. Call UseFindCloudletPerformanceMode(true)
         /// </summary>
         /// <returns>FindCloudletReply Task</returns>
-        public async Task<bool> FindCloudlet()
+        public async Task<bool> FindCloudlet(string dmeHost = null, uint dmePort = 0)
         {
             latestFindCloudletReply = null;
-
+            Debug.Log("FindCloudlet to DME: " + dmeHost);
```
Comment (franklin-huang-mobiledgex) [inline Runtime/Scripts/MobiledgeXIntegrationHelper.cs:132]: you print dmeHost later in the function. don't think you need this

Comment (lgarner) [reply Runtime/Scripts/MobiledgeXIntegrationHelper.cs]: Removed nit.

File: Runtime/Scripts/MobiledgeXIntegration.cs [hunk 2]
```diff
@@ -100,22 +106,29 @@ public MobiledgeXIntegration()
...
-            bool registered = await Register();
+            bool registered = await Register(dmeHost, dmePort);
             if (!registered)
             {
+                Debug.LogError("Register Failed!");
                 return false;
             }
-
-            return await FindCloudlet();
+            Debug.LogError("Register OK!");
```
Comment (ah1053) [inline Runtime/Scripts/MobiledgeXIntegration.cs:117]: Debug.Log not Debug.LogError

File: Runtime/Scripts/MobiledgeXIntegrationHelper.cs [hunk 3]
```diff
@@ -75,26 +75,46 @@ public async Task<bool> Register()
...
+                Debug.Log("RegisterClient Exception: " + e.Message);
+                throw e;
             }
-            if (reply.status != ReplyStatus.RS_SUCCESS)
+            finally
             {
-                throw new RegisterClientException("Bad RegisterClient. RegisterClient status is " + reply.status);
+                if (reply == null)
+                {
+                    Debug.Log("Register reply NULL!");
```
Comment (ah1053) [inline Runtime/Scripts/MobiledgeXIntegrationHelper.cs:110]: Debug.LogError

File: Runtime/Scripts/MobiledgeXIntegrationHelper.cs [hunk 4]
```diff
@@ -75,26 +75,46 @@ public async Task<bool> Register()
...
+                    throw new RegisterClientException("RegisterClient returned null.");
+                }
+                if (reply.status != ReplyStatus.RS_SUCCESS)
+                {
+                    Debug.Log("Register Failed: " + reply.status);
```
Comment (ah1053) [inline Runtime/Scripts/MobiledgeXIntegrationHelper.cs:115]: Debug.LogError

File: Runtime/Scripts/MobiledgeXIntegrationHelper.cs [hunk 5]
```diff
@@ -122,32 +140,55 @@ public async Task<bool> FindCloudlet()
             }
             catch (CarrierInfoException cie)
             {
+                Debug.Log("FindCloudlet CarrierInfoException: " + cie.Message);
```
Comment (ah1053) [inline Runtime/Scripts/MobiledgeXIntegrationHelper.cs:143]: Debug.LogError

File: Runtime/Scripts/MobiledgeXIntegrationHelper.cs [hunk 6]
```diff
@@ -75,26 +75,46 @@ public async Task<bool> Register()
             }
             catch (CarrierInfoException cie)
             {
+                Debug.Log("Register Exception: " + cie.Message);
```
Comment (ah1053) [inline Runtime/Scripts/MobiledgeXIntegrationHelper.cs:78]: Debug.LogError

File: Runtime/Scripts/MobiledgeXIntegrationHelper.cs [hunk 7]
```diff
@@ -75,26 +75,46 @@ public async Task<bool> Register()
...
             {
+                Debug.Log("RegisterClient HttpException: " + httpe.Message);
                 throw new RegisterClientException("RegisterClient Exception: " + httpe.Message + ", HTTP StatusCode: " + httpe.HttpStatusCode + ", API ErrorCode: " + httpe.ErrorCode + "\nStack: " + httpe.StackTrace);
             }
-
-            if (reply == null)
+            catch (Exception e)
             {
-                throw new RegisterClientException("RegisterClient returned null.");
+                Debug.Log("RegisterClient Exception: " + e.Message);
```
Comment (ah1053) [inline Runtime/Scripts/MobiledgeXIntegrationHelper.cs:103]: Debug.LogError

**Comment by lgarner** [comment]
Logging, yes, we can certainly improve it, including existing ones that do
what you mention here. We can discuss in another bug.

**Pull request merged**